### PR TITLE
Instant converter

### LINKED ISSUE #214: Support DATETIME type for tarantool >= 2.10

### PR DESCRIPTION
**Pull request by anatennis**
I haven't forgotten about:
- [x] Tests
- [x] Changelog
- [x] Documentation
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:
Closes #214

### CODE REVIEW

**Review by akudiyar — CHANGES_REQUESTED** [review]
Thanks for the patch! LGTM, but there are a few more things not to forget about:

- Additional tests are needed with a version switch for `ConvertersWithClusterClientIT.java` and  `ConvertersWithProxyClientIT.java` test suites
- We need to add a CI pipeline for Tarantool 2.10 for these tests

File: src/main/java/io/tarantool/driver/mappers/converters/value/DefaultExtensionValueToInstantConverter.java [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+import java.nio.ByteBuffer;
+import java.time.Instant;
+
+/**
+ * Default {@link ExtensionValue} to {@link java.time.Instant} converter
+ *
+ * @author Anastasiia Romanova
+ */
+public class DefaultExtensionValueToInstantConverter implements ValueConverter<ExtensionValue, Instant> {
```
Comment (akudiyar) [inline src/main/java/io/tarantool/driver/mappers/converters/value/DefaultExtensionValueToInstantConverter.java:14]: Please add `serialVersionUID` value here too

**Comment by anatennis** [comment]
> Thanks for the patch! LGTM, but there are a few more things not to forget about:
> 
> * Additional tests are needed with a version switch for `ConvertersWithClusterClientIT.java` and  `ConvertersWithProxyClientIT.java` test suites
> * We need to add a CI pipeline with Tarantool 2.10 for these tests

added

File: src/main/java/io/tarantool/driver/mappers/converters/object/DefaultInstantToExtensionValueConverter.java [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+ * @author Anastasiia Romanova
+ */
+public class DefaultInstantToExtensionValueConverter implements ObjectConverter<Instant, ExtensionValue> {
+
+    private static final long serialVersionUID = 20220418L;
+
+    private static final byte DATETIME_TYPE = 0x04;
+
+    private byte[] toBytes(Instant value) {
+        ByteBuffer buffer = ByteBuffer.wrap(new byte[16]);
```
[the remark this answers is not in the document]
Comment (anatennis) [reply src/main/java/io/tarantool/driver/mappers/converters/object/DefaultInstantToExtensionValueConverter.java]: yes, 8 will be better

File: src/test/java/io/tarantool/driver/integration/ConvertersWithClusterClientIT.java [hunk 3]
```diff
@@ -65,6 +66,23 @@ public void test_boxSelect_shouldReturnTupleWithUUID() throws Exception {
         Assertions.assertEquals(uuid, fields.getUUID("uuid_field"));
     }
 
+    @Test
+    @EnabledIf("io.tarantool.driver.TarantoolUtils#versionWithInstant")
+    public void test_boxSelect_shouldReturnTupleWithInstant() throws Exception {
+        //given
+        Instant instant = Instant.now();
+        client.space("space_with_instant")
```
Comment (akudiyar) [inline src/test/java/io/tarantool/driver/integration/ConvertersWithClusterClientIT.java:74]: Hmm, I don't see any changes in the Lua backend with this new space and also a CI pipeline with 2.10 for testing this. Would you mind adding them?

**Review by akudiyar — CHANGES_REQUESTED** [review]

**Comment by ArtDu** [comment]
Will be done in https://github.com/tarantool/cartridge-java/pull/376

**Pull request closed without merging**